src/classes/RD_RecurringDonations_TEST.cls

```diff
@@ -469,26 +469,354 @@ public class RD_RecurringDonations_TEST {
+
+        System.assertEquals(4, postUpdateOpps.size(),'Two new opportunities should have been created.');
+
+        for(Integer i; i<originalOpps.size(); i++) {
+            System.assertEquals(originalOpps[i].Id, postUpdateOpps[i].Id, 'The original opportunity records should persist.');
+        }
+
+        for(Integer i=0; i<postUpdateOpps.size(); i++) {
+            System.assertEquals(system.label.npe03.RecurringDonationStageName,postUpdateOpps[i].StageName, 'All opportunities should still be open.');
+        }
+
+    }
+
+    static testMethod void updateFixedLengthRecurringDonationIncreaseWithClosedOpps() {
+        if (strTestOnly != '*' && strTestOnly != 'updateFixedLengthRecurringDonationIncreaseWithClosedOpps') return;
+
+        UTIL_CustomSettingsFacade.getRecurringDonationsSettingsForTest(
+                new npe03__Recurring_Donations_Settings__c(
+                        // npe03__Opportunity_Forecast_Months__c = 12,
+                        npe03__Maximum_Donations__c = 50,
+                        npe03__Open_Opportunity_Behavior__c = RD_RecurringDonations.RecurringDonationCloseOptions.Mark_Opportunities_Closed_Lost.name()
+                ));
+
+        String closedStage = UTIL_UnitTestData_TEST.getClosedWonStage();
+
+        Account a = new Account();
+        a.Name = 'test Individual';
+        insert a;
+
+        Contact c = new Contact();
+        c.FirstName = 'test';
+        c.LastName = 'contact';
+        c.AccountId = a.Id;
+        insert c;
+
+        npe03__Recurring_Donation__c r1 = new npe03__Recurring_Donation__c();
+        r1.Name = 'test';
+        r1.npe03__Installments__c = 2;
+        r1.npe03__Contact__c = c.Id;
+        r1.npe03__Amount__c = 100;
+        r1.npe03__Installment_Period__c = system.label.npe03.RecurringDonationInstallmentPeriodWeekly;
+        r1.npe03__Date_Established__c = system.today().toStartOfMonth();
+        r1.npe03__Schedule_Type__c = system.label.npe03.RecurringDonationMultiplyValue;
+        r1.npe03__Open_Ended_Status__c = null;
+        r1.npe03__Next_Payment_Date__c = system.today().toStartOfMonth();
+        insert r1;
+
+        List<Opportunity> originalOpps = [SELECT Id
+                , Name
+                , Amount
+                , AccountId
+                , CloseDate
+                , StageName
+        FROM Opportunity
+        WHERE npe03__Recurring_Donation__c = :r1.id];
+
+        List<Error__c> errors = [SELECT Id, Full_Message__c FROM Error__c];
+        System.assertEquals(errors.size(),0,errors);
+
+        System.assertEquals(2, originalOpps.size(), 'Two opportunities should be created initially.');
+        System.assertEquals(100, originalOpps[0].Amount);
+        Test.startTest();
+        originalOpps[0].StageName = closedStage;
+        update originalOpps[0];
+        r1.npe03__Installments__c = 4;
+        update r1;
+        Test.stopTest();
+
+        errors = [SELECT Id, Full_Message__c FROM Error__c];
```

Comment:
I see a number of these queries throughout the class as well. This could be refactored into a method call
`errors = getErrorRecords();` or even something like `assertNoErrors();` where that method does the query and asserts the result is zero records.